File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.java
import java.util.*;

class Solution {
    public long StringToInt(String word){
        long result = 0;
        for(int i=word.length()-1;i>=0;i--){
            result += Math.pow(26,(word.length()-1)-i)*(long)(word.charAt(i)-96);
        }
        return result;
    }
    public String solution(long n, String[] bans) {
        String answer = "";
        TreeSet<Long> set = new TreeSet<>();
        for(int i=0;i<bans.length;i++){
            set.add(StringToInt(bans[i]));
        }
        
        if(set.size()>0){
            Long search = set.first();
            while(search<=n){
                n++;
                search = set.higher(search);
                if(search==null) break;
            }
        }
        int j = 0;
        while(n>0){
            long num = n%26;
            n = n/26;
            if(num==0) {
                answer = 'z' + answer;
                n--;
            }
            else answer = (char)(num+96) + answer;
        }
        return answer;
    }
}

File: 프로그래머스/3/389481. 봉인된 주문/봉인된 주문.js
function solution(n, bans) {
    var answer = '';
    // 26개
    // 1 -> a
    // 10 -> aa
    // 30 -> 14
    // a -> 97
    // z -> 122
    
    //console.log(a.charCodeAt(0))
    //console.log(String.fromCharCode(122))
    bans.sort((a,b)=>{
        if(a.length===b.length) return a.localeCompare(b)
        else return a.length - b.length
    })
    for(const ban of bans){
        let ban_length = 0;
        for(let i=0;i<ban.length;i++){
            ban_length += (Math.pow(26,ban.length - i - 1)*(ban.charCodeAt(i) - 96))
        }
        if(ban_length<=n) n++
    }

    if(n>26){
        while(n>0){
            n--;
            const digit = n % 26;
            answer = String.fromCharCode(97 + digit) + answer;
            n = Math.floor(n / 26);
        }
    }else{
        answer = String.fromCharCode(n+96)
    }
    return answer;
}